tools/misc/tpm2_checkquote.c

```diff
@@ -216,6 +219,37 @@ static bool pcrs_from_file(const char *pcr_file_path,
     return result;
 }
 
+static bool eventlog_from_file(tpm2_eventlog_ctx_t * ctx, const char *file_path) {
+
+    unsigned long size;
+
+    if (!files_get_file_size_path(file_path, &size)) {
+        return false;
+    }
+
+    if (!size) {
+        LOG_ERR("The eventlog file \"%s\" is empty", file_path);
+        return false;
+    }
+
+    uint8_t * eventlog = calloc(1, size);
+    if (!eventlog) {
+        LOG_ERR("OOM");
+        return false;
+    }
+
+    uint16_t size_tmp = size; // wtf uint16?
+    if (!files_load_bytes_from_path(file_path, eventlog, &size_tmp)) {
+        free(eventlog);
+        return false;
+    }
+
+    bool rc = parse_eventlog(ctx, eventlog, size);
```

Comment:
You could make this take a file pointer perhaps.... Then we don't need to load the whole thing up at once in memory. If you need a FILE stream from memory, don't use fmemopen. If you look through the logs you'll see it got removed, someone has some system without that. However, you can use the OPENSSL BIO routines, since we require OSSL.